url8d3c4d17f5_catseye_JaC64_tgz-pJ8-JaC64J8/src/com/dreamfabric/jac64/C64Screen.java

```diff
@@ -0,0 +1,2025 @@
+            tmp = 256;
+          } else {
+            tmp = 0;
+          }
+          collissionMask[collX + 7 - pix] =
+            collissionMask[collX + 6 - pix] = tmp;
+        }
+      } else {
+        for (int pix = 0; pix < 8; pix++) {
+          if ((data & (1 << pix)) > 0) {
+            mem[mpos + 7 - pix] = penColor;
+            collissionMask[collX + 7 - pix] = 256;
+          } else {
+            mem[mpos + 7 - pix] = bgcol;
+            collissionMask[collX + 7 - pix] = 0;
+          }
+        }
+      }
+
+      if (multiCol && extended) {
+        // Illegal mode => all black!
+        for (int pix = 0; pix < 8; pix++) {
+          mem[mpos + 7 - pix] = 0xff000000;
+        }
+      }
+
+      if (BAD_LINE_DEBUG && badLine) {
+        for (int pix = 0; pix < 8; pix += 4) {
+          mem[mpos + 7 - pix] = (mem[mpos + 7 - pix] & 0xff7f7f7f) | 0x0fff;
+        }
+      }
+    } else {
+      // -------------------------------------------------------------------
+      // Bitmap mode!
+      // -------------------------------------------------------------------
+      position = vicBase + (vc & 0x3ff) * 8 + rc;
+      if (multiCol) {
+        multiColor[0] = bgColor;
+      }
+      int vmliData = vicCharCache[vmli];
+      penColor =
+        cbmcolor[(vmliData & 0xf0) >> 4];
+      bgcol = cbmcolor[vmliData & 0x0f];
+
+      data = memory[position];
+
+      if (multiCol) {
+        multiColor[1] =
+          cbmcolor[(vmliData >> 4) & 0x0f];
+        multiColor[2] =
+          cbmcolor[vmliData & 0x0f];
+        multiColor[3] = cbmcolor[vicColCache[vmli] & 0x0f];
+
+        // Multicolor
+        int tmp;
+        for (int pix = 0; pix < 8; pix += 2) {
+          mem[mpos + 6 - pix] = mem[mpos + 7 - pix] =
+            multiColor[tmp = (data >> pix) & 3];
+          if (tmp > 0x01) {
+            tmp = 256;
+          } else {
+            tmp = 0;
+          }
+          collissionMask[collX + 7 - pix] =
+            collissionMask[collX + 6 - pix] = tmp;
+        }
+      } else {
+        // Non multicolor
+        for (int pix = 0; pix < 8; pix++) {
+          if ((data & (1 << pix)) > 0) {
+            mem[7 - pix + mpos] = penColor;
+            collissionMask[collX + 7 - pix] = 256;
+          } else {
+            mem[7 - pix + mpos] = bgcol;
+            collissionMask[collX + 7 - pix] = 0;
+          }
+        }
+      }
+
+      if (extended) {
+        // Illegal mode => all black!
+        for (int pix = 0; pix < 8; pix++) {
+          mem[mpos + 7 - pix] = 0xff000000;
+        }
+      }
+
+      if (BAD_LINE_DEBUG && badLine) {
+        for (int pix = 0; pix < 8; pix += 4) {
+          mem[mpos + 7 - pix] = (mem[mpos + 7 - pix] & 0xff3f3f3f) | 0x0fff;
+        }
+      }
+    }
+    vc++;
+    vmli++;
+  }
+
+  // -------------------------------------------------------------------
+  // Sprites...
+  // -------------------------------------------------------------------
+  private final void drawSprites() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 20. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.